### ISSUE TITLE
W1404: implicit-str-concat does not work for single-argument function calls

### ISSUE THREAD

**Issue by Timmmm** [PyCQA/pylint#10371]
### Bug description

The error here is detected:

```python
    print("a", "b" "c")
```
But here it is not:

```python
    print("a" "b" "c")
```
This is because the mistake is whitelisted when in parentheses (added in 81917bc9fa90fb129fe8b3c6c4523dc81a1356e3).

If it is too difficult to detect this case then I would suggest adding an option to *never* whitelist it. There's no reason to use this misguided feature - you can always just use `+` instead.

### Command used

```shell
pylint test.py
```

### Pylint output

```python
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.83/10, +0.17)
```

### Expected behavior

It should output

```
test.py:21:0: W1404: Implicit string concatenation found in call (implicit-str-concat)
```

### Pylint version

```shell
pylint 3.3.6
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I added the need specification label.

 Imo we should raise for:
```python
print("a" "b" "c")
```
But probably not for:
```python
print(
    "Two roads diverged in a yellow wood,\n"
    "And sorry I could not travel both\n"
    "And be one traveler, long I stood\n"
    "And looked down one as far as I could."
)
```
There's also this where we should probably raise (?)
```python
some_unknown_function(
    "Two roads diverged in a yellow wood,\n"
    "And sorry I could not travel both\n"
    "And be one traveler, long I stood\n"
    "And looked down one as far as I could.",
    [1, 2, 3, 4]
)
```
But not for this:
```python
some_unknown_function(
    (
        "Two roads diverged in a yellow wood,\n"
        "And sorry I could not travel both\n"
        "And be one traveler, long I stood\n"
        "And looked down one as far as I could."
    ),
    [1, 2, 3, 4],
)
```

**Comment by Timmmm** [comment]
> But probably not for

Why not? You can use `+` instead which makes the intention obvious and avoids the possibility of a mistake.

```
print(
    "Two roads diverged in a yellow wood,\n" +
    "And sorry I could not travel both\n" +
    "And be one traveler, long I stood\n" +
    "And looked down one as far as I could."
)
```

**Comment by Pierre-Sassoulas** [comment]
Favoring ``+`` instead of implicit string concatenation is opinionated and pytlint shouldn't take a stance on the matter. As evidenced by the opinion from the previous PR. Also Pycharm use implicit string concatenation when you type a new line, then black format it with explicit parenthesis. I thought implicit str concat would have better performances, but this is possibly neutral and a non issue:
```python
import timeit

implicit_code = '''
s = "Hello " "World"
'''
explicit_code = '''
s = "Hello " + "World"
'''

implicit_time = timeit.timeit(implicit_code, number=1_000_000_000)
explicit_time = timeit.timeit(explicit_code, number=1_000_000_000)

print(f"Implicit concatenation time: {implicit_time:.6f} seconds")
print(f"Explicit '+' concatenation time: {explicit_time:.6f} seconds")
```
Launched three time and got : 9.75/9. 91s, 9.81/10.10s, 9.81/ 9.68s

**Comment by Timmmm** [comment]
There's no performance difference because [they compile to exactly the same bytecode](https://godbolt.org/z/PWKzTzb46).

Maybe it shouldn't be the default, but there should at least be an option to ban all implicit string concatenation.

**Comment by Pierre-Sassoulas** [comment]
Nice website, I didn't know about it ! 

I'm neutral about the option, but I would keep the current behavior as the default. I still think there's a false negative with parenthesis from a function with a single args mistaken for explicit parenthesis around a voluntary implicit string concat that we also need to fix. Let's hear from other maintainers before committing to a new option.
